File: 739-daily-temperatures/daily-temperatures.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        # using Monotonic Stack
        stack=[]
        result=[0]*len(temperatures)
        for i in range(len(temperatures)):
            while stack and temperatures[i]>temperatures[stack[-1]]:
                j=stack.pop()
                result[j]=i-j
            stack.append(i)
        return result    






        # time limit execeed
        # result=[0]*len(temperatures)
        # for i in range(len(temperatures)):
        #     days=1
        #     find=False
        #     for j in range(i+1,len(temperatures)):
        #         if temperatures[j]>temperatures[i]:
        #             result[i]=days
        #             find=True
        #             break
        #         else:
        #             days=days+1 
        #     if find==False:
        #         result[i]=0
        # return result               


        

File: 739-daily-temperatures/daily-temperatures.java
class Solution {
    public int[] dailyTemperatures(int[] temperatures) {
        Stack<Integer> stack=new Stack<>();
        int result[]=new int[temperatures.length];
        for(int i=0;i<temperatures.length;i++){
            while(stack.size()>0 && temperatures[stack.peek()]<temperatures[i]){
                int j=stack.pop();
                result[j]=i-j;
            }
            stack.push(i);
        }
        return result;
        
    }
}